### PR TITLE
Modify Query to have classPath as tagged value when we save it via different query workflows

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

#2006

## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: .changeset/little-flies-unite.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-extension-dsl-data-space': patch
```
Comment (xannem) [inline .changeset/little-flies-unite.md:2]: Not sure if this should be separated into a different changeset or not 😓

**Review by xannem — DISMISSED** [review]

File: packages/legend-graph/src/graphManager/action/query/QueryUtils.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+import { QueryTaggedValue } from './Query.js';
+
+// NOTE: these are non-compilable profile and tag that we come up with for query
+// so that it records the class it is created from
+export const QUERY_PROFILE_PATH = 'meta::pure::profiles::query';
+export const QUERY_PROFILE_TAG_CLASS = 'class';
+
+export const createQueryClassTaggedValue = (
```
Comment (akphi) [inline packages/legend-graph/src/graphManager/action/query/QueryUtils.ts:24]: I think this is specific to query created from data-space, can we move this to where we handle the same synthetic tagged values logic for data-space query?

File: packages/legend-application-query/src/stores/QueryEditorStore.ts [hunk 3]
```diff
@@ -634,6 +635,11 @@ export class MappingQueryCreatorStore extends QueryEditorStore {
         query.groupId = this.groupId;
         query.artifactId = this.artifactId;
         query.versionId = this.versionId;
+        query.taggedValues = [
+          createQueryClassTaggedValue(
```
Comment (akphi) [inline packages/legend-application-query/src/stores/QueryEditorStore.ts:639]: I actually think we don't want to do this analytics for query not created from a dataspace :( let's not do that.

File: .changeset/little-flies-unite.md (hunk 1 above)
Comment (akphi) [reply .changeset/little-flies-unite.md:2]: I think, the change in this PR should only happen in `dsl-data-space`

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-extension-dsl-data-space/src/DSL_DataSpace_Const.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@
  */
 
 // NOTE: these are non-compilable profile and tag that we come up with for query
-// so that it records the dataSpace it is created from
+// so that it records the dataSpace and class it is created from
```
Comment (akphi) [inline packages/legend-extension-dsl-data-space/src/DSL_DataSpace_Const.ts:18]:
```suggestion
// to store extra metadata
```

**Review by akphi — CHANGES_REQUESTED** [review]

**Review by akphi — APPROVED** [review]
